PubSub/src/Topic.php

```diff
@@ -486,8 +524,16 @@ public function publish($message, array $options = [])
      */
     public function publishBatch(array $messages, array $options = [])
     {
+        $totalMessagesSize = 0;
         foreach ($messages as &$message) {
             $message = $this->formatMessage($message);
+            $totalMessagesSize += strlen(serialize($message));
```

Comment:
Can you check which of the approaches should best represent the message size for checking against the threshold?

1. `strlen(serialize($message))` 
OR
2. `strlen($message->serializeToString())`

Reply:
@vishwarajanand 

Thanks a lot for pointing this out. I check and found that the difference in size grows when we take into account multiple messages instead of a singe one. 

To rectify this logic, I've changes the implementation (and thus the testing strategy) where the compression deciding logic is now placed in the Grpc layer.

I've updated the design doc too with the decisions taken to arrive at this strategy. Please take a look [there](https://docs.google.com/document/d/1fHiCE9erqD5nM72T6Z_LATqMM6r5NAvnZyQM34BK0Q4/edit?disco=AAAAzskX2cE) too.